contracts/src/registry/ETHRegistrar.sol

```diff
@@ -0,0 +1,214 @@
+        price = prices.price(name, uint256(expiry), duration);
+    }    
+
+
+    /**
+     * @dev Make a commitment for a name.
+     * @param name The name to commit.
+     * @param owner The address of the owner of the name.
+     * @param secret The secret of the name.
+     * @param subregistry The registry to use for the commitment.
+     * @param flags The flags to use for the commitment.
+     * @param duration The duration of the commitment.
+     * @return The commitment.
+     */
+    function makeCommitment(
+        string memory name,
+        address owner,
+        bytes32 secret,
+        address subregistry,
+        uint96 flags,
+        uint64 duration
+    ) public pure override returns (bytes32) {        
+        return
+            keccak256(
+                abi.encode(
+                    name,
+                    owner,
+                    secret,
+                    subregistry,
+                    flags,
+                    duration
+                )
+            );
+    }
+
+
+    /**
+     * @dev Commit a commitment.
+     * @param commitment The commitment to commit.
+     */
+    function commit(bytes32 commitment) public override {
+        if (commitments[commitment] + maxCommitmentAge >= block.timestamp) {
+            revert UnexpiredCommitmentExists(commitment);
+        }
+        commitments[commitment] = block.timestamp;
+    }
+
+
+
+    /**
+     * @dev Register a name.
+     * @param name The name to register.
+     * @param owner The owner of the name.
+     * @param subregistry The subregistry to register the name in.
+     * @param flags The flags to set on the name.   
+     * @param duration The duration of the registration.
+     * @return tokenId The token ID of the registered name.
+     */
+    function register(
+        string calldata name,
+        address owner,
+        IRegistry subregistry,
+        uint96 flags,
+        uint64 duration
+    ) external payable onlyRole(CONTROLLER_ROLE) returns (uint256 tokenId) {
+        IPriceOracle.Price memory price = rentPrice(name, duration);
+        uint256 totalPrice = price.base + price.premium;
+        if (msg.value < totalPrice) {
+            revert InsufficientValue();
+        }
+
+        _consumeCommitment(name, duration, makeCommitment(name, owner, bytes32(0), address(subregistry), flags, duration));
+
+        tokenId = registry.register(name, owner, subregistry, flags, uint64(block.timestamp) + duration);
+
+        if (msg.value > totalPrice) {
+            payable(msg.sender).transfer(msg.value - totalPrice);
+        }
+
+        emit NameRegistered(name, owner, subregistry, flags, duration, tokenId);
+    }
+
+    /**
+     * @dev Renew a name.
+     * @param name The name to renew.
+     * @param duration The duration of the renewal.
+     */
+    function renew(string calldata name, uint64 duration) external payable onlyRole(CONTROLLER_ROLE) {
```

Comment:
```suggestion
    function renew(string calldata name, uint64 duration) external payable {
```

Reply:
If it's publicly callable what's the purpose of the controler role then?